File: 1313. Decompress Run-Length Encoded List/solution.py
class Solution(object):
  def decompressRLElist(self, nums):
    """
    :type nums: List[int]
    :rtype: List[int]
    """
    result = []
    for index in range(0, len(nums) - 1, 2):
      frequency = nums[index]
      value = nums[index + 1]
      for amount in range(frequency):
        result.append(value)

    return result

File: 1313. Decompress Run-Length Encoded List/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var decompressRLElist = function(nums) {
  let result = [];
  for (let index = 0; index < nums.length - 1; index += 2){
    let frequency = nums[index];
    let value = nums[index + 1];
    for (let amount = 0; amount < frequency; amount++){
      result.push(value);
    }
  }

  return result;
};